examples/StoreApi/StoreValue.cs

```diff
@@ -0,0 +1,49 @@
+namespace Momento.Sdk.StoreClient;
+
+public enum StoreValueType
+{
+    String,
+    Integer64,
+    Double,
+    Bool
+}
+
+public class StoreValue
+{
+    private object _value;
+    public StoreValueType Type { get; }
+
+    public StoreValue(object value, StoreValueType type)
+    {
+        _value = value;
+        Type = type;
+    }
+
+    public StoreValue(string value) : this(value, StoreValueType.String)
+    {
+    }
+
+    public StoreValue(long value) : this(value, StoreValueType.Integer64)
+    {
+    }
+
+    public StoreValue(double value) : this(value, StoreValueType.Double)
+    {
+    }
+
+    public StoreValue(bool value) : this(value, StoreValueType.Bool)
+    {
+    }
+
+    // implicit cast from the various types into StoreValue
+    public static implicit operator StoreValue(string value) => new StoreValue(value);
```

Comment:
theoretically the right-hand-side of these could be a code block, correct? Allowing us to put in some logging or throw a different error type if we decide to, etc.?

Not saying we should change it from what you have here, just wanted to understand our options.

Reply:
Yes. The right-hand side here is just a shorthand when the function body is a single statement. We can have a multi-statement block instead.